source/macos/AdaptiveCards/AdaptiveCards/Views/ACRContentStackView.swift

```diff
@@ -316,26 +324,39 @@ class ACRContentStackView: NSView, ACRContentHoldingViewProtocol, SelectActionHa
     /// activation constraint all at once is more efficient than activating
     /// constraints one by one.
     
-    func configureLayoutAndVisibility(_ verticalContentAlignment: ACSVerticalContentAlignment, minHeight: NSNumber?, heightType: ACSHeightType, type: ACSCardElementType, isRootMinHeightAvailable: Bool) {
-        self.verticalContentAlignment = verticalContentAlignment
+    func configureLayoutAndVisibility(element: NSObject, isRootMinHeightAvailable: Bool) {
+        var minHeight: NSNumber?
+        var isBackgroundImageAvailable = false
+        if let cardElement = element as? ACSAdaptiveCard {
+            self.verticalContentAlignment = cardElement.getVerticalContentAlignment()
+            minHeight = cardElement.getMinHeight()
+            let bgImage = cardElement.getBackgroundImage()
+            if let imageUrl = bgImage?.getUrl(), URL(string: imageUrl) != nil {
+                isBackgroundImageAvailable = true
+            }
+        } else if let collectionElement = element as? ACSCollectionTypeElement {
+            self.verticalContentAlignment = collectionElement.getVerticalContentAlignment()
+            minHeight = collectionElement.getMinHeight()
+            let bgImage = collectionElement.getBackgroundImage()
+            if let imageUrl = bgImage?.getUrl(), URL(string: imageUrl) != nil {
+                isBackgroundImageAvailable = true
+            }
+        }
```

Comment:
can we move this `isBackgroundImageAvailable` at caller function. so we can remove the whole element dependency from this methods?
you can create tuple that can have necessary card properties.

```
typealias ACSElementProperties = (verticalContentAlignment: ACSVerticalContentAlignment, minHeight: NSNumber?, heightType: ACSHeightType, type: ACSCardElementType, isBackgroundImageAvailable: Bool)
```

```
func configureLayoutAndVisibility(elementProperties: ACSElementProperties, isRootMinHeightAvailable: Bool) {

}
```